agenta-cli/agenta/sdk/middleware/vault.py

```diff
@@ -108,23 +111,24 @@ async def _get_secrets(self, request: Request) -> Optional[Dict]:
 
         vault_secrets: List[SecretDTO] = []
 
-        try:
-            async with httpx.AsyncClient() as client:
-                response = await client.get(
-                    f"{self.host}/api/vault/v1/secrets",
-                    headers=headers,
-                )
+        if not _UNAUTHORIZED_EXECUTION_ALLOWED:
+            try:
+                async with httpx.AsyncClient() as client:
+                    response = await client.get(
+                        f"{self.host}/api/vault/v1/secrets",
+                        headers=headers,
+                    )
 
-                if response.status_code != 200:
-                    vault_secrets = []
+                    if response.status_code != 200:
+                        vault_secrets = []
 
-                else:
-                    secrets = response.json()
-                    vault_secrets = self._transform_secrets_response_to_secret_dto(
-                        secrets
-                    )
-        except:  # pylint: disable=bare-except
-            display_exception("Vault: Vault Secrets Exception")
+                    else:
+                        secrets = response.json()
+                        vault_secrets = self._transform_secrets_response_to_secret_dto(
+                            secrets
+                        )
+            except:  # pylint: disable=bare-except
+                display_exception("Vault: Vault Secrets Exception")
```

Comment:
[Ditto](https://github.com/Agenta-AI/agenta/pull/2443/files#r1937586698). For this, it was trying to fetch secrets from the vault before getting the local secrets.